Content.Server/Cargo/Systems/CargoSystem.Bounty.cs

```diff
@@ -420,6 +420,10 @@ public bool TryAddBounty(EntityUid uid, CargoBountyPrototype bounty, StationCarg
             return false;
 
         _nameIdentifier.GenerateUniqueName(uid, BountyNameIdentifierGroup, out var randomVal);
+        var newBounty = new CargoBountyData(bounty, randomVal);
+        // This bounty id already exists! Probably because NameIdentifierSystem ran out of ids.
+        if (component.Bounties.Any(b => b.Id == newBounty.Id))
+            return false;
```

Comment:
```suggestion
        if (component.Bounties.Any(b => b.Id == newBounty.Id))
        {
            Log.Error("Failed to add bounty {ID} because another one with the same ID already existed!", newBounty.Id);
            return false;
        }
```